solr/core/src/java/org/apache/solr/cloud/api/collections/Assign.java

```diff
@@ -562,8 +566,36 @@ public static AssignStrategy createAssignStrategy(CoreContainer coreContainer) {
         coreContainer.getPlacementPluginFactory().createPluginInstance();
     if (placementPlugin == null) {
       // Otherwise use the default
-      // TODO: Replace this with a better options, such as the AffinityPlacementFactory
-      placementPlugin = (new SimplePlacementFactory()).createPluginInstance();
+      String defaultPluginId = System.getProperty(DEFAULT_PLACEMENT_PLUGIN_SYSPROP);
```

Comment:
Why not pass the fully qualified classname in the system property so that Assign.java does not have to be touched to use a new placement plugin? Security concerns?

Reply:
If we have SolrResourceLoader available that should not be too hard, unless you also need to support loading plugins from solr packages, which (unfortunately) uses a custom loader system with need for package prefixes.

Reply:
Keeping it simple in this first iteration, postponing class loading until someone asks for it :)